### DISCUSSION TITLE
NPM behind SSO provider (Authentik)

### DISCUSSION THREAD

**Discussion by mirisbowring** [NginxProxyManager/nginx-proxy-manager#2386] [Q&A]
Hi,

I just wanted to know if NPM supports Basic Auth or an authentication header?
I am using Authentik in front of most of my applications as my SSO Provider.

I've seen that v3 will contain LDAP but that does not support SSO, which is why i would like to use authentik to provide login into NPM

**Comment by whiteonred** [comment]
I also use Authentik for all my services and still haven't found a simple solution for logging into NPM. Sad.

**Comment by jc21** [comment]
v3 - which was a golang rewrite - is no longer on the board. Though it did have Oauth and ldap authentication options, they are yet to be implemented properly in the nodejs backend. There are a few PR's from the community already but I haven't had the time to dissect them. On the surface they seem to be doing too much or overcomplicating the implementation.
